packages/number-field/src/NumberField.ts

```diff
@@ -510,6 +512,7 @@ export class NumberField extends TextfieldBase {
                     value -= this.step;
                 }
             }
+            value = parseFloat(this.valueFormatter.format(value));
```

Comment:
Is it possible to only format at the end to reduce the chances of fidelity loss?